api/mapping/views.py

```diff
@@ -287,24 +287,99 @@ def get_queryset(self):
             return ScanReport.objects.all().distinct()
 
         return ScanReport.objects.filter(
+            # parent dataset and SR are public checks
             Q(
-                parent_dataset__project__members=self.request.user.id,
+                # parent dataset and SR are public
                 parent_dataset__visibility=VisibilityChoices.PUBLIC,
```

Comment:
Does this mean if the Dataset and SR are public, that it doesn't matter whether you're a member of a Project linked to the Dataset? Or is there a filter on Project membership before this?

Reply:
I moved that check down to the bottom of the query since it's the same for all cases. Because it acts like a `kwarg` while the big `Q` statement is an `arg`, it has to go last.